packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_web.dart

```diff
@@ -14,16 +14,66 @@
 
 import 'dart:async';
 
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
 import 'package:amplify_secure_storage_dart/src/exception/not_available_exception.dart';
 import 'package:amplify_secure_storage_dart/src/exception/secure_storage_exception.dart';
-import 'package:amplify_secure_storage_dart/src/interfaces/amplify_secure_storage_interface.dart';
-import 'package:amplify_secure_storage_dart/src/interfaces/secure_storage_interface.dart';
 import 'package:amplify_secure_storage_dart/src/js/indexed_db.dart';
+import 'package:amplify_secure_storage_dart/src/platforms/amplify_secure_storage_in_memory.dart';
 
 /// The web implementation of [SecureStorageInterface].
 class AmplifySecureStorageWeb extends AmplifySecureStorageInterface {
   AmplifySecureStorageWeb({required super.config});
 
+  /// The [SecureStorageInterface] instance to use.
+  late final Future<SecureStorageInterface> _instance = () async {
+    if (config.webOptions.persistenceOption == WebPersistenceOption.inMemory) {
+      return const AmplifySecureStorageInMemory();
+    }
+    if (indexedDB == null) {
+      logger.warn(
+        'IndexedDB is not available. '
+        'Falling back to in-memory storage.',
```

Comment:
I'm assuming that most developers will understand the implication of this, but if we get GH issues to the contrary we should consider adding a statement making it explicit.